examples/weather/regen/example_inference.py

```diff
@@ -0,0 +1,209 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#!/usr/bin/env python
+# coding: utf-8
+
+
+import sys
+
+sys.path.append("./training")  # for loading pickled models which need training/utils
+import numpy as np
+import torch
+from sda.score import GaussianScore_from_denoiser, VPSDE_from_denoiser, VPSDE
+from training.utils.diffusions.networks import get_preconditioned_architecture
+import torch
+import xarray as xr
+from scipy.stats import norm
+import matplotlib
+
+matplotlib.rcParams.update({"font.size": 14})
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+from configs.base import (
+    path_to_model_state,
+    path_to_hrrr,
+    isd_path,
+    station_locations,
+    path_to_pretrained,
+)
+import scipy.interpolate
+
+
+def bounds():
+    dy = lat_g[0] - lat_g[1]
+    dx = lon_g[1] - lon_g[0]
+    north_west = [lat_g[0] + dy / 2, lon_g[0] - dx / 2]
+    south_east = [lat_g[-1] - dy / 2, lon_g[-1] + dx / 2]
+    return [north_west, south_east]
+
+
+def interpolate(x):
+    x = np.asarray(x)
+
+    values = x.ravel()
+    values_g = scipy.interpolate.griddata(
+        (lat.ravel(), lon.ravel()), values, (lat_g[:, None], lon_g)
+    )
+    return values_g
+
+
+u10_mean, u10_std = -0.262, 2.372
+v10_mean, v10_std = 0.865, 4.115
+logtp_mean, logtp_std = -8.117, 2.489
+
+
+means, stds = np.array([u10_mean, v10_mean, logtp_mean]), np.array(
+    [u10_std, v10_std, logtp_std]
+)
+
+
+print("cuda available? ", torch.cuda.is_available())
+
+
+## Load Obs data
+
+
+ds_regrid = xr.open_dataset(isd_path)
```

Comment:
I thought we had discussed hardcoding a list of obs here, so that users can inference w/o the large netCDF full of NaNs.

Reply:
I thought we were leaving that for later refactoring